draft-ietf-ppm-dap.md

```diff
@@ -2789,16 +2860,26 @@ of this task:
 
 The Leader then combines the values inside the batch bucket as follows:
 
-* Aggregate shares are combined via `Vdaf.merge(agg_param, agg_shares)` ((see
-  {{!VDAF, Section 4.4}})), where `agg_param` is the aggregation parameter
+* Aggregate shares are combined via `Vdaf.merge(agg_param, agg_shares)` (see
+  {{!VDAF, Section 4.4}}), where `agg_param` is the aggregation parameter
   provided in the `CollectionJobReq`, and `agg_shares` are the (partial)
   aggregate shares in the batch buckets. The result is the Leader aggregate
   share for this collection job.
 * Report counts are combined via summing.
 * Checksums are combined via bitwise XOR.
 
-Then the Leader sends a POST request to
-`{helper}/tasks/{task-id}/aggregate_shares` with the following message:
+A Helper aggregate share is identified by a 16-byte ID:
+
+~~~ tls-presentation
+opaque AggregateShareID[16];
+~~~
+
+The Helper's aggregate share is an HTTP resource served by the Helper at the URL
+`{helper}/tasks/{task-id}/aggregate_shares/{aggregate-share-id}`. To obtain it,
+the Leader first chooses an aggregate share ID, which MUST be unique within the
+scope of the corresponding DAP task.
+
+Then the Leader sends a PUT request to the aggregate share with the body:
```

Comment:
The aggregate share ID is new, and we need it so that the leader can poll on something for asynchronous agg share handling. The alternative would be to not specify an aggregate share ID and let the helper choose a unique URL for the aggregate share, but we chose long ago to have HTTP clients choose resource identifiers (i.e. leader chooses agg job and agg share IDs, collector chooses collection job IDs).

Reply:
We discussed this at the last PPM interim. What I proposed was to let the leader choose it and nobody argued against that ([1]). My rationale was that we should be consistent with all the other interactions where HTTP clients always choose identifiers (clients choose report IDs, leaders choose aggregation job IDs, collectors choose collection job IDs).

[1]: https://datatracker.ietf.org/meeting/interim-2025-ppm-01/materials/slides-interim-2025-ppm-01-sessa-tim-geoghegan-aligning-dap-with-rfc-9205-00